### PR TITLE
feat: support to communicate with etcd by TLS in command line

### PR DESCRIPTION
**Pull request by tokers**
### What this PR does / why we need it:

This PR implements the SSL handshaking to etcd cluster in the APISIX command line tool, and it will obey the TLS configuration in `config.yaml`.

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -54,7 +54,7 @@ jobs:
           submodules: recursive
 
       - name: Linux Get dependencies
-        run: sudo apt install -y cpanminus build-essential libncurses5-dev libreadline-dev libssl-dev perl
+        run: sudo apt install -y cpanminus build-essential libncurses5-dev libreadline-dev libssl-dev perl libssl-dev
```
Comment (membphis) [inline .github/workflows/build.yml:57]: why we need this? only for testing? or the apisix need it?

Comment (Yiyiyimu) [reply .github/workflows/build.yml:57]: It seems duplicate? The original install list already got `libssl-dev`

**Comment by membphis** [comment]
ci failed: https://github.com/apache/apisix/pull/3415/checks?check_run_id=1762297012

File: .github/workflows/build.yml (hunk 1 above)
Comment (tokers) [reply .github/workflows/build.yml]: Removed.

File: doc/install-dependencies.md [hunk 2]
```diff
@@ -53,7 +53,7 @@ sudo yum install yum-utils
 sudo yum-config-manager --add-repo https://openresty.org/package/centos/openresty.repo
 
 # install OpenResty and some compilation tools
-sudo yum install -y openresty curl git gcc luarocks lua-devel
+sudo yum install -y openresty curl git gcc luarocks lua-devel openresty-openssl-devel
```
Comment (spacewander) [inline doc/install-dependencies.md:56]: Look like we should also install luarocks from the source?
We can do it with utils/linux-install-luarocks.sh, but we need to move the `OPENSSL` variable configuration to this script.

Comment (tokers) [reply doc/install-dependencies.md:56]: OK.

Comment (tokers) [reply doc/install-dependencies.md:56]: @spacewander This is the document for installing dependencies.

Comment (tokers) [reply doc/install-dependencies.md:56]: I think we could hint the users to install LuaRocks 3.x, and add the operations for OPENSSL variables here.

File: doc/install-dependencies.md [hunk 3]
```diff
@@ -55,6 +55,15 @@ sudo yum-config-manager --add-repo https://openresty.org/package/centos/openrest
 # install OpenResty and some compilation tools
 sudo yum install -y openresty curl git gcc luarocks lua-devel openresty-openssl-devel
 
+# if the luarocks installed by yum is not the 3.x, installing it from source.
+sudo bash ./utils/linux-install-luarocks.sh
```
Comment (spacewander) [inline doc/install-dependencies.md:59]: Maybe we need to use some wget/curl trick to get the script.

Comment (tokers) [reply doc/install-dependencies.md]: Changed.

File: doc/install-dependencies.md (hunk 2 above)
Comment (spacewander) [reply doc/install-dependencies.md:56]: Need to remove luarocks from this list.

File: doc/install-dependencies.md [hunk 4]
```diff
@@ -73,7 +78,10 @@ tar -xvf etcd-v3.4.13-linux-amd64.tar.gz && \
     sudo cp -a etcd etcdctl /usr/bin/
 
 # install OpenResty and some compilation tools
-sudo yum install -y openresty curl git gcc luarocks lua-devel
+sudo yum install -y openresty curl git gcc luarocks lua-devel openresty-openssl-devel
```
Comment (spacewander) [inline doc/install-dependencies.md:81]: Ditto

File: doc/install-dependencies.md (hunk 2 above)
Comment (tokers) [reply doc/install-dependencies.md:56]: Fixed.

File: doc/install-dependencies.md (hunk 3 above)
Comment (tokers) [reply doc/install-dependencies.md]: Fixed.

File: doc/install-dependencies.md (hunk 4 above)
Comment (tokers) [reply doc/install-dependencies.md]: Fixed.

**Review by spacewander — APPROVED** [review]

File: Makefile [hunk 5]
```diff
@@ -52,8 +53,16 @@ help: default
 .PHONY: deps
 deps: default
 ifeq ($(LUAROCKS_VER),luarocks 3.)
-	luarocks install --lua-dir=$(LUAJIT_DIR) rockspec/apisix-master-0.rockspec --tree=deps --only-deps --local
+	mkdir ~/.luarocks || true
+	luarocks config variables.OPENSSL_LIBDIR $(addprefix $(OR_PREFIX), openssl/lib)
+	luarocks config variables.OPENSSL_INCDIR $(addprefix $(OR_PREFIX), openssl/include)
+	luarocks install rockspec/apisix-master-0.rockspec --tree=deps --only-deps --local
 else
+	@echo "WARN: You're not using LuaRocks 3.x, please add the following items to your LuaRocks config file:"
```
Comment (membphis) [inline Makefile:61]: nice warning

**Review by membphis — COMMENTED** [review]
we need to fix https://github.com/apache/apisix/issues/3461 issue first.

this PR may fail after we used `luajit` in CLI.

**Comment by tokers** [comment]
> we need to fix #3461 issue first.
> 
> this PR may fail after we used `luajit` in CLI.

Happy to do adjusting after the #3461 fixed.

**Review by Yiyiyimu — APPROVED** [review]
LGTM

File: .travis/linux_tengine_runner.sh [hunk 6]
```diff
@@ -194,6 +194,13 @@ tengine_install() {
     cp -r ${OPENRESTY_PREFIX}/* build-cache${OPENRESTY_PREFIX}
     ls build-cache${OPENRESTY_PREFIX}
     rm -rf openresty-${OPENRESTY_VERSION}
+
```
Comment (moonming) [inline .travis/linux_tengine_runner.sh:197]: do we still support tengine now?

**Comment by Yiyiyimu** [comment]
Hi @tokers since #3461 got fixed, do you mind merge master and see how it works

File: utils/linux-install-luarocks.sh [hunk 7]
```diff
@@ -16,11 +16,19 @@
 # limitations under the License.
 #
```
Comment (moonming) [inline utils/linux-install-luarocks.sh:18]: are we not support luarocks 2.x?

**Comment by tokers** [comment]
> Hi @tokers since #3461 got fixed, do you mind merge master and see how it works

OK, I'll.

File: utils/linux-install-luarocks.sh (hunk 7 above)
Comment (tokers) [reply utils/linux-install-luarocks.sh:18]: I think LuaRocks 3.x is better to use?

**Comment by tokers** [comment]
> > Hi @tokers since #3461 got fixed, do you mind merge master and see how it works
> 
> OK, I'll.

Updated.

File: utils/linux-install-luarocks.sh (hunk 7 above)
Comment (moonming) [reply utils/linux-install-luarocks.sh:18]: agreed. So Apache APISIX not support luarocks 2.x, right?

**Pull request merged**